lib/rules/function-calc-no-unspaced-operator/index.mjs

```diff
@@ -85,134 +85,47 @@ const rule = (primary, _secondaryOptions, context) => {
 
 			if (!FUNC_CALLS_REGEX.test(value)) return;
 
-			let needsFix = false;
-			const valueIndex = declarationValueIndex(decl);
-
-			const tokens = tokenize({ css: value });
-
-			{
-				// Step 1
-				// Step 1.1
-				// Re-tokenize dimensions with units containing dashes.
-				// These might be typo's.
-				// For example: `10px-20px` has a unit of `px-20px`
-				tokens.forEach((token, i) => {
-					if (!isTokenDimension(token)) return;
-
-					const { unit } = token[4];
-
-					if (unit.startsWith('--')) return;
-
-					const indexOfDash = unit.indexOf('-');
-
-					if (indexOfDash === -1) return;
-
-					const remainder = unit.slice(indexOfDash);
-
-					if (remainder.length === 1) return;
-
-					mutateUnit(token, unit.slice(0, indexOfDash));
-					token[3] = token[2] + token[1].length;
-
-					const remainderTokens = tokenize({ css: remainder }).slice(0, -1); // Trim EOF token
-
-					remainderTokens.forEach((remainderToken) => {
-						remainderToken[2] += token[3];
-						remainderToken[3] += token[3];
-					});
-
-					tokens.splice(i + 1, 0, ...remainderTokens);
-				});
-
-				// Step 1.2
-				// Re-tokenize scss interpolation blocks
-				// Grouping `#` and `{` into a single token allows us to parse these as simple blocks with curly braces.
-				// For example: `#{$foo}`
-				tokens.forEach((currentToken, i) => {
-					if (!isTokenDelim(currentToken) || currentToken[4].value !== '#') return;
-
-					const nextToken = tokens[i + 1];
-
-					if (!isTokenOpenCurly(nextToken)) return;
-
-					const nextNextToken = tokens[i + 2];
-
-					if (!isTokenDelim(nextNextToken) || nextNextToken[4].value !== '$') return;
-
-					// Set the string representation of the open curly to `#{`
-					nextToken[1] = '#{';
-					// Remove the `#` token
-					tokens.splice(i, 1);
-				});
-			}
-
-			const nodes = parseListOfComponentValues(tokens);
+			const nodes = tokenizeDeclarationValue(value);
 
 			if (nodes.length === 0) return;
 
+			const valueIndex = declarationValueIndex(decl);
+			const fixDeclarationValue = () => setDeclarationValue(decl, stringify([nodes]));
+
 			/**
 			 * @param {ContainerNode} node
-			 * @param {CompleteOperation} operation
+			 * @param {Operation} operation
 			 */
 			function checkCompleteOperation(node, operation) {
 				if (operation.before.length && operation.after.length) return;
 
-				if (!operation.before.find(isWhitespaceNode)) {
-					if (context.fix) {
-						needsFix = true;
-						node.value.splice(
-							node.value.indexOf(operation.operator),
-							0,
-							new WhitespaceNode([[TokenType.Whitespace, ' ', -1, -1, undefined]]),
-						);
-					} else {
-						complain(
-							messages.expectedBefore(operation.operatorToken[4].value),
-							decl,
-							valueIndex + operation.operatorToken[2],
-							operation.operatorToken[4].value,
-						);
-					}
+				if (!operation.before.some(isWhitespaceNode)) {
```

Comment:
suggestion
refactor

```js
/** @param {keyof messages} method */
const foo = (method) => {
	const position = method === 'expectedAfter' ? 'after' : 'before';

	if (!operation[position].some(isWhitespaceNode)) {
		complain(
			messages[method](operation.operatorChar),
			decl,
			valueIndex + operation.operatorCharPosition,
			operation.operatorChar,
			() => {
				operation.insertWhitespace(node, position);
				fixDeclarationValue();
			},
		);
	}
};
```

I am not sure if it's necessary though.
It depends on the clarity of the function name chosen.

Reply:
Good catch. Improved via 6944be586.